server.go

```diff
@@ -490,7 +472,7 @@ func (s *Server) shutdown() {
 			return false
 		}
 
-		if c.consID != "" {
+		if c.consumer != nil {
```

Comment:
For the shake of encapsulation, it'd be better to add a method here, `Client.Consumer() *Consumer` and use that instead of accessing an unexported struct field.

Reply:
Ack! I'll update the PR accordingly.

Reply:
Unfortunately this change is not that straightforward. Since the `ConsumerID` contains all the topics in which the `Consumer` is subscribed to, we can't simply call the `Consumer` upon a `RPUSH` request for a multi-topic `Consumer`, since offsets commits operation only run per single topic.

I agree with your rational, however I'd suggest to handle it separately in a subsequent PR.